app/src/main/java/chat/rocket/android/authentication/domain/model/DeepLinkInfo.kt

```diff
@@ -0,0 +1,94 @@
+package chat.rocket.android.authentication.domain.model
+
+import android.annotation.SuppressLint
+import android.content.Intent
+import android.net.Uri
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+import timber.log.Timber
+
+// see https://rocket.chat/docs/developer-guides/deeplink/ for documentation
+
+@SuppressLint("ParcelCreator")
+@Parcelize
+data class DeepLinkInfo(
+    val url: String,
+    val userId: String?,
+    val token: String?,
+    val rid: String?,
+    val roomType: String?,
+    val roomName: String?
+) : Parcelable
+
+fun Uri.getDeepLinkInfo(): DeepLinkInfo? {
+    return if (isAuthenticationDeepLink()) {
+        val host = getQueryParameter("host")
+        val url = if (host.startsWith("http")) host else "https://$host"
```

Comment:
Why are you prepending "https" to all of your url values?  Why not just use the "scheme" value that comes with the data?

Reply:
This code was in the original LoginDeepLinkInfo. It might be improved if I called the variable serverUrl. This is the URL used to connect to the RC server - so it can't use a scheme like "rocketchat".

Reply:
I get that, thanks.  But in this case, what is the original 'scheme' that comes with the data?  Is that 'rocketchat' in this case?

Reply:
the lines you are commenting on are in the original code. In fact, IsAuthenticationDeepLink is defined exactly as in the original code. This is all that was implemented prior to my changes. This code handles both "rocketchat" scheme and "http" scheme but fundamentally this code path is unchanged by me. I also agree that it is a little confusing.